### PR TITLE
support v-on passive modifier

### PR DESCRIPTION
**Pull request by Kingwl**
- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### CODE REVIEW

File: src/core/util/env.js [hunk 1]
```diff
@@ -16,6 +16,18 @@ export const isAndroid = UA && UA.indexOf('android') > 0
...
+if (inBrowser) {
+  try {
+    const opts = {}
+    Object.defineProperty(opts, 'passive', ({
+      get: function () { _supportsPassive = true }
+    } : Object)) // https://github.com/facebook/flow/issues/285
+    window.addEventListener('test-passive', null, opts)
+  } catch (e) {}
+}
+export const supportsPassive = _supportsPassive
```
Comment (yyx990803) [inline src/core/util/env.js]: ES export bindings track changes, we can just do

``` js
export let supportsPassive = false
// feature detect...
```

File: test/unit/features/directives/on.spec.js [hunk 2]
```diff
@@ -526,4 +526,28 @@ describe('Directive v-on', () => {
     expect(spyUp.calls.count()).toBe(1)
     expect(spyDown.calls.count()).toBe(1)
   })
+
+  it('should support passive', () => {
```
Comment (yyx990803) [inline test/unit/features/directives/on.spec.js]: This test case should only run when the test browser supports passive.

File: src/core/util/env.js (hunk 1 above)
Comment (Kingwl) [inline src/core/util/env.js]: done

File: test/unit/features/directives/on.spec.js (hunk 2 above)
Comment (Kingwl) [inline test/unit/features/directives/on.spec.js]: done

**Comment by yyx990803** [comment]
Looking good! I'll merge this when we do the next minor release.

**Comment by Kingwl** [comment]
#5118

**Comment by HerringtonDarkholme** [comment]
Just a suggestion, since Vue also support `prevent` modifier https://vuejs.org/v2/guide/events.html#Event-Modifiers. It would even better if compiler can detect and warn the mutual exclusive `prevent` and `passive`.

**Comment by Kingwl** [comment]
ping
@HerringtonDarkholme @yyx990803 
review need

File: src/compiler/helpers.js [hunk 3]
```diff
@@ -41,6 +42,12 @@ export function addHandler (
   modifiers: ?ASTModifiers,
   important: ?boolean
 ) {
+  // warn prevent and passive modifier
+  if (process.env.NODE_ENV !== 'production' && modifiers && modifiers.prevent && modifiers.passive) {
+    warn(
+      'prevent and passive are mutual exclusive.'
```
Comment (HerringtonDarkholme) [inline src/compiler/helpers.js]: hmmm, what about `passive and prevent can't be used together. Passive handler can't prevent default  event.`

It uses simpler word and explains the reason.

Comment (Kingwl) [inline src/compiler/helpers.js]: thanks

File: src/compiler/helpers.js [hunk 4]
```diff
@@ -41,6 +42,13 @@ export function addHandler (
   modifiers: ?ASTModifiers,
   important: ?boolean
 ) {
+  // warn prevent and passive modifier
+  if (process.env.NODE_ENV !== 'production' && modifiers && modifiers.prevent && modifiers.passive) {
+    warn(
+      'passive and prevent can\t be used together. ' +
```
Comment (jbergstroem) [inline src/compiler/helpers.js]: missing literal (`'`) in `can\t`

File: test/unit/features/directives/on.spec.js [hunk 5]
```diff
@@ -526,4 +527,36 @@ describe('Directive v-on', () => {
...
+      vm.$refs.normal.checked = false
+      vm.$refs.passive.checked = false
+      vm.$refs.exclusive.checked = false
+      vm.$refs.normal.click()
+      vm.$refs.passive.click()
+      vm.$refs.exclusive.click()
+      expect(vm.$refs.normal.checked).toBe(false)
+      expect(vm.$refs.passive.checked).toBe(true)
+      expect(vm.$refs.exclusive.checked).toBe(true)
+      expect('passive and prevent can\t be used together. Passive handler can\'t prevent default event.').toHaveBeenWarned()
```
Comment (jbergstroem) [inline test/unit/features/directives/on.spec.js]: needs minor fix for above suggested text change.

**Comment by znck** [comment]
@Kingwl could you look into it? There is an error after updating to `dev` branch.

**Pull request merged**